### ISSUE TITLE
Uncovered files from dirs without __init__.py are not reported

### ISSUE THREAD

**Issue by blueyed** [pytest-dev/pytest-cov#88]
It seems like files that are not imported/used are not reported as having 0% coverage.

It would be useful if all the files of the project would be in the coverage report - having a percentage of 0% in case they were not imported/used.

I could imagine that pytest-cov would have to hook into py.test's collection process for this, and then somehow make coverage aware of them - even if they are not imported during tests?!

**Comment by ionelmc** [comment]
This feature is builtin into coverage, but with a caveat: you have to use paths for the source.

Eg: if you use `py.test --cov=somepath` then you'll get all files in `somepath` in the report.

**Comment by blueyed** [comment]
And that's not covered by `[paths] source = X` and/or `[run] source = X`?

**Comment by ionelmc** [comment]
Not really sure, but it sounds like it's not something pytest-cov specific. What behavior do you get if you use plain coverage (eg: `coverage run -mpytest tests`)?

If you put an example somewhere I'll take a look.

**Comment by codingjoe** [comment]
:+1: same here

**Comment by ionelmc** [comment]
@codingjoe are you having the same issue or you're waiting too on some example that reproduces the issue? :grin:

**Comment by codingjoe** [comment]
Never mind problem gone. I had the same problem too, but only once. Now it's gone, very curious. Made my coverage skyrocket. Now it's back to "normal".

**Comment by ionelmc** [comment]
@blueyed Can you create some minimal project that reproduce the issue? Or at least point to the project that has this problem ...

**Comment by codingjoe** [comment]
I can't even recreate the issue myself. So no, sorry.

**Comment by blueyed** [comment]
This happens because of a missing `__init__.py` file.

coveragepy skips it then (https://github.com/nedbat/coveragepy/blob/e7c532c2dbcef6281df5a7a0db53eca4083a026a/coverage/files.py#L342-L346).

This gets used in `Coverage.get_data` (https://github.com/nedbat/coveragepy/blob/e7c532c2dbcef6281df5a7a0db53eca4083a026a/coverage/control.py#L823-L831).

I think that also "non-importable" files should be included - but that is an issue to raise for coveragepy then probably?!

In my case the `__init__.py` file was missing in `project/app/management/__init__.py`, but that is no problem for Django itself - the commands can be used.

**Comment by ionelmc** [comment]
Do we have a bug report for this in https://bitbucket.org/ned/coveragepy?

**Comment by blueyed** [comment]
I could not find one, and given the comment in https://github.com/nedbat/coveragepy/blob/e7c532c2dbcef6281df5a7a0db53eca4083a026a/coverage/files.py#L342-L346 it might even be not possible?!

**Comment by junaruga** [comment]
Though I am not sure my case is same case with this report.
I was succeeded to report files in directory without `__init__.py` in my case.

```
$ cat .coveragerc 
# https://coverage.readthedocs.io/en/latest/config.html
[run]
include =
    rpm_py_installer/*
    install.py
    setup.py
```

```
$ pytest --cov-config .coveragerc --cov . tests
...
Name                           Stmts   Miss  Cover
--------------------------------------------------
install.py                       325     27    92% <=== This file is not in __init__.py's directory, and reported here.
rpm_py_installer/__init__.py       0      0   100%
rpm_py_installer/version.py        1      0   100%
setup.py                          35     35     0%
--------------------------------------------------
TOTAL                            361     62    83%
```

See https://github.com/junaruga/rpm-py-installer/pull/59 for detail.

**Comment by ionelmc** [comment]
Closing this in favour of #176. Add your thoughts there if you want your usecase documented.

**Comment by remort** [comment]
Problem with modules placed in dirs without init files is still exists, while
https://github.com/pytest-dev/pytest-cov/issues/176 is closed.

The bug is exists. In python 3 modules (.py files) doesn't need init diles nearby to be able to be imported. I don't use init files and always import directly from .py files. The whole lotta aplications written without ini files and pytest do bot cover such dirs (if they are not imported or tested somehow in tests runned by pytest).

**Comment by bhanuunrivalled** [comment]
what a mess

**Comment by sylvain-ri** [comment]
The current solution is to use `include_namespace_packages  = True` 
https://coverage.readthedocs.io/en/7.0.0/config.html#report-include-namespace-packages

**Comment by sylvain-ri** [comment]
For people also ending here, there is the solution:
add a config file for your pytest-cov command `pytest --cov=<folder> --cov-config=.coverage_rc`

Content of `.coverage_rc`
```
[report]
include_namespace_packages = true

# might also be convenient to add one of these:
omit = **/__init__.py
skip_empty = true
```

**Comment by sylvain-ri** [comment]
For people also ending here, there is the solution:
add a config file for your pytest-cov command `pytest --cov=<folder> --cov-config=.coverage_rc`

Content of `.coverage_rc`
```
[report]
include_namespace_packages = true

# might also be convenient to add one of these:
omit = **/__init__.py
skip_empty = true
```
Adapt the heading to `[tool.coverage.report]` if within a pyproject.toml file
